scripts/quests/sandoria/Trouble_at_the_Sluice.lua

```diff
@@ -0,0 +1,120 @@
+
+quest.reward =
+{
+    fame = 30,
+    fameArea = xi.quest.fame_area.SANDORIA,
+    item = xi.items.HEAVY_AXE,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE and
+                player:getFameLevel(xi.quest.fame_area.SANDORIA) >= 3 and
+                player:hasCompletedQuest(xi.quest.log_id.SANDORIA, xi.quest.id.sandoria.THE_RUMOR)
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Belgidiveau'] = quest:progressEvent(57),
+
+            onEventFinish =
+            {
+                [57] = function(player, csid, option, npc)
+                    if option == 0 then
+                        quest:begin(player)
+                        quest:setVar(player, 'Prog', 1)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.NORTHERN_SAN_DORIA] =
+        {
+            ['Belgidiveau'] =
+            {
+                onTrigger = function(player, npc)
+                    if not player:hasKeyItem(xi.ki.NEUTRALIZER) then
+                        return quest:event(55)
+                    elseif player:hasKeyItem(xi.ki.NEUTRALIZER) then
+                        return quest:progressEvent(56)
+                    end
+                end,
+
+            },
+
+            onEventFinish =
+            {
+                [56] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:delKeyItem(xi.ki.NEUTRALIZER)
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.BOSTAUNIEUX_OUBLIETTE] =
+        {
+            ['Novalmauge'] =
+            {
+
+                onTrade = function(player, npc, trade)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        npcUtil.tradeHasExactly(trade, xi.items.DAHLIA)
+                    then
+                        return quest:progressEvent(17)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(15)
+                    elseif quest:getVar(player, 'Prog') == 2 then
+                        return quest:event(16)
+                    else
+                        return quest:event(10)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [15] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+
+                [17] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 0)
```

Comment:
Strange setting the prog back to 0 in order to progress the quest lol. Makes readability a bit backwards. As well as troubleshooting for GMs

Reply:
ah i didnt delete it - good catch.